funsor/tensor.py

```diff
@@ -695,6 +696,18 @@ def eager_getitem_tensor_tensor(op, lhs, rhs):
     return Tensor(data, inputs, lhs.dtype)
 
 
+# TODO handle variable length Tuple
+@eager.register(Finitary, Op, typing.Tuple[Tensor])
```

Comment:
Note the use of `typing.Tuple` in head patterns like this is not currently supported, and will have to be worked around like we did with `Contraction`.  This is what got me thinking about #355, #433 and #434 again.

Reply:
Added support for this in #451.